sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala

```diff
@@ -178,6 +257,47 @@ case class FlatMapGroupsWithStateExec(
       }
     }
 
+    /**
+     * Process the new data iterator along with the initial state. The initial state is applied
+     * before processing the new data for every key. The user defined function is called only
+     * once on the data.
+     */
+    def processNewDataWithInitState(
+        childDataIter: Iterator[InternalRow],
+        initStateIter: Iterator[InternalRow]
+      ): Iterator[InternalRow] = {
+
+      if (!childDataIter.hasNext && !initStateIter.hasNext) return Iterator.empty
+
+      val groupedChildDataIter = GroupedIterator(childDataIter, groupingAttributes, child.output)
+      val groupedInitStateIter =
+        GroupedIterator(initStateIter, initStateGroupAttrs, initialState.output)
+
+      val keyOrderingComparator = GenerateOrdering.generate(
+        groupingAttributes.map(SortOrder(_, Ascending)), groupingAttributes)
```

Comment:
this object `groupingAttributes.map(SortOrder(_, Ascending))` must be reused between here and the `requiredChildOrdering`.  the code has to ensure that the child ordering done by the SparkPlan and the comparator being used for this merging ... is based on the same sorting strategy.